datatypes/src/jvmTest/kotlin/Asn1TimeTest.kt

```diff
@@ -0,0 +1,39 @@
+import at.asitplus.crypto.datatypes.asn1.Asn1Time
+import io.kotest.core.spec.style.FreeSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import kotlinx.datetime.toKotlinInstant
+import java.time.Instant
+import java.util.*
+
+class Asn1TimeTest : FreeSpec ({
+    val now = Date.from(Instant.now())
+    val then = Date.from(Instant.now().plusSeconds(500L))
+    val later = Date.from(Instant.now().plusSeconds(824046715L))
+    val asn1Time = Asn1Time(now.toInstant().toKotlinInstant())
+    val asn1Time1 = Asn1Time(then.toInstant().toKotlinInstant())
+    val asn1Time2 = Asn1Time(then.toInstant().toKotlinInstant(), Asn1Time.Format.UTC)
+    val asn1Time3 = Asn1Time(later.toInstant().toKotlinInstant(), Asn1Time.Format.GENERALIZED)
```

Comment:
this is probably enough, but of course property-based testing would look much nicer. The actual testing logic is very thorough, though! fine for me